dataRetriever/shardedData/shardedData_test.go

```diff
@@ -41,6 +42,29 @@ func TestNewShardedData_GoodConfigShouldWork(t *testing.T) {
 	sd, err := shardedData.NewShardedData(cacheConfigBad)
 	assert.Nil(t, err)
 	assert.NotNil(t, sd)
+	assert.False(t, sd.IsInterfaceNil())
```

Comment:
check.IfNil and remove assert on line 44

Reply:
refactored.